.github/workflows/ci.yml

```diff
@@ -16,4 +16,13 @@ jobs:
   python-checks:
     name: Python Checks
     uses: ./.github/workflows/python-checks.yml
+    permissions:
+      id-token: write
+      contents: read
 
+  wheels:
```

Comment:
Do we want to run this on every PR/ push on main? Beyond adding the changes in main, what do you have in mind in terms of release workflow?

Reply:
I think yes. We need to run integration tests before merging to main, but that step is part of building wheels. We need to build wheel to be able to run integration tests.

Reply:
We need to split them if thats the case. Integ-tests should run even when we are not building wheels. It should also run, with wheels, after we build them. But building wheels should not be a dependency of integration-testings. If we split these two concerns, we then probably want Integ-Tests to run on every PR and push to main. Wheels only build on Tags and Wheel build step is followed by a set if Integ-tests running with the artefacts of that step.